src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -821,147 +828,145 @@ public unsafe void CopyTo(Array array, int index)
             }
             else if (array is bool[] boolArray)
             {
-                if (array.Length - index < m_length)
+                // This method uses unsafe code to manipulate data in the BitArray.  To avoid issues with
+                // buggy code concurrently mutating this instance in a way that could cause memory corruption,
+                // we snapshot the array then operate only on this snapshot.  We don't care about such code
+                // corrupting the BitArray data in a way that produces incorrect answers, since BitArray is not meant
+                // to be thread-safe; we only care about avoiding buffer overruns.
```

Comment:
I know it is copied from the existing code, but in fact it doesn't protect from buffer overruns as m_array and m_length have to be queries under a lock for that, otherwise overruns are still possible

Reply:
Later it checks `if (thisLength < 0 || thisArray.Length < GetInt32ArrayLengthFromBitLength(thisLength))` to make sure `thisArray` and `thisLength` are reasonable.